src/contentmanagerdelegate.cpp

```diff
@@ -285,10 +285,28 @@ void ContentManagerDelegate::handleLastColumnClicked(const QModelIndex& index, Q
 
 QSize ContentManagerDelegate::sizeHint(const QStyleOptionViewItem &option, const QModelIndex &index) const
 {
-    Q_UNUSED(option);
-
-    if (isDescriptionIndex(index)) {
-        return QSize(300, 70);
+    if (isDescriptionIndex(index))
+    {
+        auto header = KiwixApp::instance()->getContentManager()->getView()->getView()->header();
+
+        /* Get the width from start of header section 1 to 4. This is a
+           slight underestimate of the actual width, but it means we will produce 
+           a proper height upper bound from boundingRect and the painting process 
+           will span the width regardless of what we return in sizeHint.
+        */
+        int width = header->length() - header->sectionSize(0) - header->sectionSize(5);
+        QRect descRect(0, 0, width, 0);
+
+        /* Based on the rectangle and text, find the best fitting size. */
+        QFontMetrics fm(option.font);
+        const QString text = index.data().toString();
+        const auto format = Qt::AlignLeft | Qt::AlignVCenter | Qt::TextWordWrap;
+        const QRect textRect = fm.boundingRect(descRect, format, text);
+        QSize textSize = textRect.size();
+
+        if (textSize.height() < 70)
+            textSize.setHeight(70);
+        return textSize;
```

Comment:
This still doesn't work well.

After a little experimenting myself I have arrived at the following code that seems to do the job:

```c++
        const auto treeView = KiwixApp::instance()->getContentManager()->getView()->getView();

        const int width = treeView->header()->length() - 2*treeView->indentation();
        // XXX: see QTreeView::padding in resources/css/_contentManager.css
        const int verticalPadding = 4;
        const int horizontalPadding = 4;
        QRect descRect(0, 0, width - 2 * horizontalPadding, 0);

        /* Based on the rectangle and text, find the best fitting size. */
        QFontMetrics fm(option.font);
        const QString text = index.data().toString();
        const auto format = Qt::AlignLeft | Qt::AlignVCenter | Qt::TextWordWrap;
        const int textHeight = fm.boundingRect(descRect, format, text).height();
        return QSize(width, std::max(textHeight + verticalPadding, 70));
```

Reply:
Done